pootle/apps/pootle_store/syncer.py

```diff
@@ -0,0 +1,365 @@
+        self.set_unitid(newunit)
+        self.set_context(newunit)
+        self.set_obsolete(newunit)
+        return newunit
+
+    def create_unit(self, unitclass):
+        return unitclass(self.source)
+
+    def set_developer_notes(self, newunit):
+        notes = self.developer_notes
+        if notes:
+            newunit.addnote(notes, origin="developer")
+
+    def set_fuzzy(self, newunit):
+        newunit.markfuzzy(self.isfuzzy)
+
+    def set_unitid(self, newunit):
+        newunit.setid(self.unitid)
+
+    def set_context(self, newunit):
+        newunit.setcontext(self.context)
+
+    def set_locations(self, newunit):
+        locations = self.locations
+        if locations:
+            newunit.addlocations(locations)
+
+    def set_obsolete(self, newunit):
+        if self.isobsolete:
+            newunit.makeobsolete()
+
+    def set_notes(self, newunit):
+        self.set_developer_notes(newunit)
+        self.set_translator_notes(newunit)
+
+    def set_target(self, newunit):
+        newunit.target = self.target
+
+    def set_translator_notes(self, newunit):
+        notes = self.translator_notes
+        if notes:
+            newunit.addnote(notes, origin="translator")
+
+
+class StoreSyncer(object):
+    unit_sync_class = UnitSyncer
+
+    def __init__(self, store):
+        self.store = store
+
+    @cached_property
+    def disk_store(self):
+        return self.store.file.store
+
+    @property
+    def translation_project(self):
+        return self.store.translation_project
+
+    @property
+    def language(self):
+        return self.translation_project.language
+
+    @property
+    def project(self):
+        return self.translation_project.project
+
+    @property
+    def source_language(self):
+        return self.project.source_language
+
+    @property
+    def store_file_path(self):
+        return os.path.join(
+            self.translation_project.abs_real_path,
+            self.store.name)
+
+    @cached_property
+    def file_class(self):
+        # get a plugin adapted file_class
+        fileclass = format_classes.gather().get(
+            str(self.store.filetype.extension))
+        if fileclass:
+            return fileclass
+        if self.store.is_template:
+            # namedtuple is equiv here of object() with name attr
+            return self._getclass(
+                namedtuple("instance", "name")(
+                    name=".".join(
+                        [os.path.splitext(self.store.name)[0],
+                         str(self.store.filetype.extension)])))
+        return self._getclass(self.store)
+
+    def convert(self, fileclass=None):
+        """export to fileclass"""
+        fileclass = fileclass or self.file_class
+        logging.debug(
+            u"Converting %s to %s",
+            self.store.pootle_path,
+            fileclass)
+        output = fileclass()
+        output.settargetlanguage(self.language.code)
+        # FIXME: we should add some headers
```

Comment:
Probably here `update_store_header` should be called. Then leave to target format to do their thing.